docs/source/architecture/injecting_batches_block_validation_rules.rst

```diff
@@ -0,0 +1,150 @@
+setting will hold a set of validation rules that are enforced for each block.
+The setting key where the validation rules are stored is
+sawtooth.validator.block_validation_rules and it will contain a string
+representation of the additional rules to enforce. The on-chain rules will be
+enforced by the block validator.
+
+A simple syntax will be used for specifying validations rules as defined by
+the following:
+
+- Individual rules shall consist of the rule name following by a ":" followed
+  by a list of arguments the rule takes.
+- If a rule takes more than one argument, the individual arguments in the list
+  shall be separated by ",".
+- If multiple rules are specified, they shall be separated by ";".
+- Spaces, tabs, and newlines are ignored
+
+The following rules are defined:
+
+NofX
+  Only N of transaction type X may be included in a block. The first argument
+  must be interpretable as an integer. The second argument is interpreted as
+  the name of a transaction family. For example, the string "NofX:2,intkey"
+  means only allow 2 intkey transactions per block.
+
+XatY
+  A transaction of type X must be in the block at position Y. The first argument
+  is interpreted as the name of a transaction family. The second argument must
+  be interpretable as an integer and defines the index of the transaction in
+  the block that must be checked. Negative numbers can be used and count
+  backwards from the last transaction in the block. The first transaction in the
+  block has index 0. The last transaction in the block has index -1. If abs(Y)
+  is larger than the number of transactions per block, then there would not be
+  a transaction of type X at Y and the block would be invalid. For example, the
+  string "XatY:intkey,0" means the first transaction in the block must be an i
+  ntkey transaction.
+
+local
+  A transaction must be signed by the same key as the block. This rule takes a
+  list of transaction indices in the block and enforces the rule on each. This
+  rule is useful in combination with the other rules to ensure a client is not
+  submitting transactions that should only be injected by the winning validator.
+
+BlockInfoInjector
+=================
+
+The BlockInfoInjector interface automatically injects BlockInfo transactions
+into the beginning of blocks.
+
+Partial example of a BlockInfoInjector implementation that will inject a single
+BlockInfo transaction at the beginning of every block:
+
+.. code-block:: python
+
+  class BlockInfoInjector(BatchInjector):
+  	def __init__(self, block_store, state_view_factory):
+      	    self._block_store = block_store
+      	    self._state_view_factory = state_view_factory
+
+  	def block_start(self, previous_block_id):
+  	    block = self._block_store[previous_block_id]
+      	    state_view = self._state_view_factory.create_view(block.state_root_hash)
+      	    block_info_batch = _create_block_info_batch(state_view)
+      	    return [block_info_batch]
```

Comment:
Do other architecture sections contain class definitions?  If not, we may want to replace the code with an explanation of what it does.

It may also make sense to have the block info stuff be an example section. What do you think?